### PR TITLE
Add support for Firefox for iOS

### PR DESCRIPTION
**Pull request by flodolo**

### CODE REVIEW

**Comment by flodolo** [comment]
Notes for testing: need to 
* Point the l10n webservice to a beta version that includes firefox-ios
```
l10nwebservice = "https://l10n.mozilla-community.org/~flod/mozilla-l10n-query-beta/"
```
* Run `setup.sh` to generate sources and folders
* Run `glossaire.sh` for at least one locale to generate the TMX

Notes for deploying: in the end this code can go out to production whenever we're ready, Firefox for iOS will only be available when it's added to the official webservice.

**Comment by flodolo** [comment]
Not ready for review: missing parts (links in search results), clash badly with mozilla.org moving to GitHub.

**Comment by flodolo** [comment]
Rebased and fixed the missing links, should be good for review.

File: app/classes/Transvision/AnalyseStrings.php [hunk 1]
```diff
@@ -44,7 +44,11 @@ public static function differences($tmx_source, $tmx_target, $repo, $ignored_str
...
             $patterns = [
-                'l10njs'     => '/\{\{\s*([a-z0-9_]+)\s*\}\}/iu', // {{foobar2}}
+                'l10njs' => '/\{\{\s*([a-z0-9_]+)\s*\}\}/iu', // {{foobar2}}
+            ];
+        } elseif ($repo == 'firefox_ios') {
+            $patterns = [
+                'ios' => '/(%(?:[0-9]+\$){0,1}@)/i', // %@, but also %1$@, %2$@, etc.
             ];
         } else {
             $patterns = [
```
Comment (pascalchevrel) [inline app/classes/Transvision/AnalyseStrings.php]: nit: I am not a fan of multiple if/elseif/else statements, I tend to think that a switch is more readable:
```php
switch($repo) {
    case Strings::startsWith($repo, 'gaia'):
        $patterns = [
            'l10njs' => '/\{\{\s*([a-z0-9_]+)\s*\}\}/iu', // {{foobar2}}
        ];
        break;
    case 'firefox_ios':
        $patterns = [
            'ios' => '/(%(?:[0-9]+\$){0,1}@)/i', // %@, but also %1$@, %2$@, etc.
        ];
        break;
    default:
        $patterns = [
            'dtd'        => '/&([a-z0-9\.]+);/i',                        // &foobar;
            'printf'     => '/(%(?:[0-9]+\$){0,1}(?:[0-9].){0,1}(S))/i', // %1$S or %S. %1$0.S and %0.S are valid too
            'properties' => '/(?<!%[0-9])\$[a-z0-9\.]+\b/i',             // $BrandShortName, but not "My%1$SFeeds-%2$S.opml"
            'l10njs'     => '/\{\{\s*([a-z0-9_]+)\s*\}\}/iu',            // {{foobar2}} Used in Loop and PDFViewer
        ];
        break;
}
```

**Comment by pascalchevrel** [comment]
Works fine, I am OK to merge it, I have a general preference for using switches instead of elseifs but I think it's really a matter of taste and shouldn't block merging this PR.

**Pull request merged**